src/entity/Pawn.java

```diff
@@ -29,35 +29,41 @@ public Move[] getValidMoves(ArrayList<Integer> position, HashMap<ArrayList<Integ
                 && (Math.abs(lastMove.getDestination().get(0) - position.get(0)) == 1)) { // and it's right next to our pawn
 
             Move enPassant = makeEnPassant(position, lastMove, moveDirection);
-            validMoves.add(enPassant);
+            possibleMoves.add(enPassant);
         }
 
         // check if our pawn can move 1 square forward
         if (checkSquare(position.get(0), position.get(1) + moveDirection, boardState).equals("empty")) {
             Move pawnPush = makePawnPush(position, moveDirection);
-            validMoves.add(pawnPush);
+            possibleMoves.add(pawnPush);
 
             // if 1 was possible and pawn hasn't moved yet, then check if moving 2 squares forward would be legal
             if (!hasMoved && checkSquare(position.get(0), position.get(1) + 2 * moveDirection, boardState).equals("empty")) {
                 Move doublePawnPush = makeDoublePawnPush(position, moveDirection);
-                validMoves.add(doublePawnPush);
+                possibleMoves.add(doublePawnPush);
             }
         }
 
```

Comment:
when adding captures to the list of moves, don't we need to update each Move object using the ".setPieceCaptured()" method, so that our view and use cases know to update accordingly when passed the moves as input data

Reply:
In this particular case, it's not possible for pawns to capture while moving forwards. I think in general, all capture moves are created appropriately, but if there's a missing case, please let me know.